system/Database/MySQLi/Connection.php

```diff
@@ -408,16 +408,19 @@ protected function _listColumns(string $table = ''): string
 	//--------------------------------------------------------------------
 
 	/**
-	 * Returns an object with field data
+	 * Returns an array of objects with field data
 	 *
 	 * @param	string	$table
-	 * @return	array
+	 * @return	\stdClass[]
+	 * @throws DatabaseException
 	 */
-	public function _fieldData(string $table)
+	public function _fieldData(string $table): array
 	{
-		if (($query = $this->query('SHOW COLUMNS FROM ' . $this->protectIdentifiers($table, TRUE, NULL, FALSE))) === FALSE)
+		$table = $this->protectIdentifiers($table, TRUE, NULL, FALSE);
+
+		if (($query = $this->query('SHOW COLUMNS FROM ' . $table)) === FALSE)
 		{
-			return FALSE;
+			throw new DatabaseException('Failed to get field data from MySQL.');
```

Comment:
Please use a localized string that is added to `system/Language/en/Database.php`